flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/CachingTableLoader.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.flink;
+
+import java.io.IOException;
+import org.apache.flink.annotation.Experimental;
+import org.apache.flink.util.Preconditions;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.util.DateTimeUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A table loader that will only reload a table after a certain interval has passed. WARNING: This
+ * table loader should be used carefully when used with writer tasks. It could result in heavy load
+ * on a catalog for jobs with many writers.
+ */
+@Experimental
+public class CachingTableLoader implements TableLoader {
```

Comment:
after looking at other classes using `Supplier<Table>`, I think this could be called `RefreshableTableSupplier` that implements `Supplier<Table>`

Reply:
I moved this to the `sink` package, made this package private, and removed `@Experimental`. I left it as a `TableLoader`. I originally had it as a `Supplier<Table>` and the feedback was that it was redundant with `TableLoader`.

Reply:
sorry. there are probably 2 reasons I changed my mind (1) I missed how `Supplier<Table> tableSupplier` was used like in `ManifestOutputFileFactory`. also  (2) earlier, `TableSupplier` was exposed as public to users in `FlinkSink`. that can be confusing to users with `TableLoader` vs `TableSupplier`. 

With it being non-public and hidden from users, I think `TableSupplier` makes more sense to me. We can pass the same object to `ManifestOutputFileFactory` and others directly. I just checked the commit history. I actually like the initial commit